Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm

```diff
@@ -3353,38 +3353,79 @@ - (CABasicAnimation *)_animationForFindOverlay:(BOOL)animateIn
+        std::max<CGFloat>(CGRectGetMinY(contentViewFrame) - minY, 0))];
+
+    [_findOverlaysOutsideContentView->right setFrame:CGRectMake(
+        CGRectGetMaxX(contentViewFrame),
+        CGRectGetMinY(contentViewFrame),
+        std::max<CGFloat>(maxX - CGRectGetMaxX(contentViewFrame), 0),
+        std::max<CGFloat>(maxY - CGRectGetMinY(contentViewFrame), 0))];
+
+    [_findOverlaysOutsideContentView->bottom setFrame:CGRectMake(
+        minX,
+        CGRectGetMaxY(contentViewFrame),
+        std::max<CGFloat>(CGRectGetMaxX(contentViewFrame) - minX, 0),
+        std::max<CGFloat>(maxY - CGRectGetMaxY(contentViewFrame), 0))];
+
+    [_findOverlaysOutsideContentView->left setFrame:CGRectMake(
+        minX,
+        minY,
+        std::max<CGFloat>(CGRectGetMinX(contentViewFrame) - minX, 0),
+        std::max<CGFloat>(CGRectGetMaxY(contentViewFrame) - minY, 0))];
 }
 
 - (void)_showFindOverlay
 {
-    if (!_findOverlay) {
-        _findOverlay = adoptNS([[UIView alloc] init]);
-        UIColor *overlayColor = [UIColor colorWithRed:(26. / 255) green:(26. / 255) blue:(26. / 255) alpha:(64. / 255)];
-        [_findOverlay setBackgroundColor:overlayColor];
-    }
+    if (!_findOverlaysOutsideContentView) {
+        auto makeOverlay = [&]() {
+            UIColor *overlayColor = [UIColor colorWithRed:(26. / 255) green:(26. / 255) blue:(26. / 255) alpha:(64. / 255)];
+            auto newOverlay = adoptNS([[UIView alloc] init]);
+            [newOverlay setBackgroundColor:overlayColor];
+            [_scrollView insertSubview:newOverlay.get() belowSubview:_contentView.get()];
 
+            return newOverlay;
+        };
+        _findOverlaysOutsideContentView = { makeOverlay(), makeOverlay(), makeOverlay(), makeOverlay() };
+    }
     [self _updateFindOverlayPosition];
-    [_scrollView insertSubview:_findOverlay.get() belowSubview:_contentView.get()];
 
     if (CALayer *contentViewFindOverlayLayer = [self _layerForFindOverlay]) {
-        [[_findOverlay layer] removeAllAnimations];
-        [contentViewFindOverlayLayer removeAllAnimations];
+        [self _updateFindOverlaysOutsideContentView:^(RetainPtr<UIView> view) {
+            [[view layer] removeAllAnimations];
+            [view setAlpha:1];
+        }];
 
-        [_findOverlay setAlpha:1];
+        [contentViewFindOverlayLayer removeAllAnimations];
         [contentViewFindOverlayLayer setOpacity:1];
     } else {
-        [_findOverlay setAlpha:0];
+        if (_findOverlaysOutsideContentView) {
+            [self _updateFindOverlaysOutsideContentView:^(RetainPtr<UIView> view) {
+                [view setAlpha:0];
+            }];
+        }
         [self _addLayerForFindOverlay];
     }
 }
 
 - (void)_hideFindOverlay
 {
     CALayer *contentViewFindOverlayLayer = [self _layerForFindOverlay];
-    CALayer *findOverlayLayer = [_findOverlay layer];
+    CALayer *findOverlayLayer = [_findOverlaysOutsideContentView.value().top layer];
```

Comment:
What guarantees `_findOverlaysOutsideContentView` is not `std::nullopt` here? I think you need to update the check of line 3430 to check for that, and then move this line after the early return – otherwise this is a change in logic.